lib/compact.js

```diff
@@ -411,11 +475,19 @@ api.compactIri = ({
       containers.push('@index');
     }
 
-    // prefer `['@graph', '@set']` and then `@graph` if value is a simple graph
-    // TODO: support `@graphId`?
-    if(_isSimpleGraph(value)) {
+    // prefer most specific container including @graph, prefering @set variations
+    if(_isGraph(value)) {
+      containers.push('@graph@index@set');
+      containers.push('@graph@index');
+      containers.push('@graph@id@set');
+      containers.push('@graph@id');
       containers.push('@graph@set');
       containers.push('@graph');
+    } else if(_isObject(value) && !_isValue(value)) {
+      containers.push('@id@set');
+      containers.push('@id');
+      containers.push('@type@set');
```

Comment:
Flipped order?

Reply:
Should be `@set@type`, and can `push` can take multiple arguments.

Based on https://github.com/json-ld/json-ld.org/pull/569 it's even more complicated, if we want to prioritize order based on the actual value, and allow `@none` to be used. The work's not all done fo this yet, but may as well get this in order.